kratos/utilities/nurbs_utilities/snake_sbm_utilities.cpp

```diff
@@ -0,0 +1,872 @@
+                        int node1_i = i;   int node1_j = j;
+                        int node2_i = i+1; int node2_j = j;
+
+                        IndexType id_node_1 = id_surrogate_first_node + node1_i + node1_j*nKnotSpans[0];
+                        IndexType id_node_2 = id_surrogate_first_node + node2_i + node2_j*nKnotSpans[0];
+                            
+                        Condition::Pointer pcond = rSurrogateModelPartInner.CreateNewCondition("LineCondition2D2N", id_surrogate_condition, {{id_node_1, id_node_2}}, p_cond_prop );
+                        // BOUNDARY true means that the condition (i.e. the sbm face) is entering looking from x,y,z positive
+                        pcond->Set(BOUNDARY, false);
+
+                        // surrogate_model_part_inner.AddCondition(p_cond);
+                        id_surrogate_condition++;
+                        check_next_point = false;
+                    } 
+                }
+                else if (knotSpansAvailable[idMatrix][j][i] == 1) {
+                    // ENTERING --> save first face in direction
+                    int node1_i = i;   int node1_j = j;
+                    int node2_i = i+1; int node2_j = j;
+
+                    IndexType id_node_1 = id_surrogate_first_node + node1_i + node1_j*(nKnotSpans[0]);
+                    IndexType id_node_2 = id_surrogate_first_node + node2_i + node2_j*(nKnotSpans[0]);
+                    Condition::Pointer pcond = rSurrogateModelPartInner.CreateNewCondition("LineCondition2D2N", id_surrogate_condition, {{id_node_1, id_node_2}}, p_cond_prop );
+                    // surrogate_model_part_inner.AddCondition(p_cond);
+                    id_surrogate_condition++;
+                    check_next_point = true;
+
+                    // BOUNDARY true means that the condition (i.e. the sbm face) is entering looking from x,y,z positive
+                    pcond->Set(BOUNDARY, true);
+
+                }
+            }
+        }
+
+        // Create "fictituos element" to store starting and ending condition id for each surrogate boundary loop
+        IndexType elem_id = rSurrogateModelPartInner.NumberOfElements()+1;
+        IndexType id_surrogate_last_condition = id_surrogate_condition-1;
+        std::vector<ModelPart::IndexType> elem_nodes{id_surrogate_first_condition, id_surrogate_last_condition};
+        rSurrogateModelPartInner.CreateNewElement("Element2D2N", elem_id, elem_nodes, p_cond_prop);
+    }
+
+
+
+    void SnakeSbmUtilities::CreateSurrogateBuondaryFromSnakeOuter(
+        std::vector<std::vector<std::vector<int>>> & knotSpansAvailable,
+        int idMatrix, 
+        ModelPart& rSurrogateModelPartOuter, 
+        ModelPart& rSkinModelPartOuter,
+        DynamicBins& testBinOuter, 
+        std::vector<int>& nKnotSpans, 
+        Vector& knotVectorU, 
+        Vector& knotVectorV, 
+        const Vector& startingPositionUV)
```

Comment:
Which is the difference of this method w.r.t. ` CreateSurrogateBuondaryFromSnakeInner`?